riscv-sbi/core/fdt/fdt-internal.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Phoenix SBI
+ *
+ * FDT internal definitions
+ *
+ * Copyright 2024 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _FDT_INTERNAL_H_
+#define _FDT_INTERNAL_H_
+
+
+#include "types.h"
+
+
+#define FDT_MAGIC      0xd00dfeed
+#define FDT_BEGIN_NODE 1
+#define FDT_END_NODE   2
+#define FDT_PROP       3
+#define FDT_NOP        4
+#define FDT_END        9
+
+
+#define FDT_EOK        0
+#define FDT_EBADOFFS   -1
+#define FDT_EBADSTRUCT -2
+#define FDT_EEND       -3
+#define FDT_ENODEEND   -4
+#define FDT_ENOTFOUND  -5
+
+
+#define fdt32_to_cpu(x) (__builtin_bswap32(x))
+#define fdt64_to_cpu(x) (__builtin_bswap64(x))
```

Comment:
Ok for now, but perhaps endianess of the cpu could be defined in `platform` module?

Reply:
May as well use the predefined `__BYTE_ORDER__` macro, will do so.